indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -227,19 +232,25 @@ public TaskStatus call()
 
                         final ProcessHolder processHolder;
                         final String childHost = node.getHost();
-                        final int childPort;
-                        final int childChatHandlerPort;
-
-                        if (config.isSeparateIngestionEndpoint()) {
-                          Pair<Integer, Integer> portPair = portFinder.findTwoConsecutiveUnusedPorts();
-                          childPort = portPair.lhs;
-                          childChatHandlerPort = portPair.rhs;
-                        } else {
-                          childPort = portFinder.findUnusedPort();
-                          childChatHandlerPort = -1;
+                        int childPort = -1;
+                        int tlsChildPort = -1;
+                        int childChatHandlerPort = -1;
+
+                        if(serverConfig.isPlainText()) {
+                          if (config.isSeparateIngestionEndpoint()) {
+                            Pair<Integer, Integer> portPair = portFinder.findTwoConsecutiveUnusedPorts();
+                            childPort = portPair.lhs;
+                            childChatHandlerPort = portPair.rhs;
+                          } else {
+                            childPort = portFinder.findUnusedPort();
+                          }
+                        }
+
+                        if(serverConfig.isTls()) {
+                          tlsChildPort = tlsPortFinder.findUnusedPort();
```

Comment:
not a blocker for this, just to confirm the expected behavior. 
it seems separateIngestionEndpoint is not supported for TLS ? In that case, can we add that to docs